src/model/core/fuel.jl

```diff
@@ -0,0 +1,146 @@
+
+@doc raw"""
+    fuel!(EP::Model, inputs::Dict, setup::Dict)
+
+This function creates expressions to account for total fuel consumption (e.g., coal, 
+natural gas, hydrogen, etc). It also has the capability to model heat rates that are
+a function of load via a piecewise-linear approximation.
+
+***** Expressions ******
+Users have two options to model the fuel consumption as a function of power generation: 
+(1). Use a constant heat rate, regardless of the minimum load or maximum load; and 
+(2). Use the PiecewiseFuelUsage-related parameters to model the fuel consumption via a 
+piecewise-linear approximation. By using this option, users can represent the fact that most 
+generators have a decreasing heat rate as a function of load.
+
+(1). Constant heat rate. 
+The fuel consumption for power generation $vFuel_{y,t}$ is determined by power generation 
+($vP_{y,t}$) mutiplied by the corresponding heat rate ($Heat\_Rate_y$). 
+The fuel costs for power generation and start fuel for a plant $y$ at time $t$, 
+denoted by $eCFuelOut_{y,t}$ and $eFuelStart$, are determined by fuel consumption ($vFuel_{y,t}$ 
+and $eStartFuel$) multiplied by the fuel costs (\$/MMBTU)
+(2). Piecewise-linear approximation
+With this formulation, the heat rate of generators becomes a function of load.
+In reality this relationship takes a nonlinear form, but we model it
+through a piecewise-linear approximation:
+
+```math
+\begin{aligned}
+vFuel_{y,t} >= vP_{y,t} * h_{y,x} + U_{g,t}* f_{y,x}
+\hspace{1cm} \forall y \in G, \forall t \in T, \forall x \in X
+\end{aligned}
+```
+Where $h_{y,x}$ represents the heat rate slope for generator $y$ in segment $x$ [MMBTU/MWh],
+ $f_{y,x}$ represents the heat rate intercept (MMBTU) for a generator $y$ in segment $x$ [MMBTU],
+and $U_{y,t}$ represents the commitment status of a generator $y$ at time $t$. These parameters
+are optional inputs to "Generators_data.csv".
+
+Since fuel consumption and fuel costs are postive, the optimization will force the fuel usage
+to be equal to the highest fuel usage segment for any given value of vP.
+When the power output is zero, the commitment variable $U_{g,t}$ will bring the intercept 
+to be zero such that the fuel consumption is zero when thermal units are offline.
+
+In order to run piecewise fuel consumption module,
+the unit commitment must be turned on, and users should provide PWFU_Slope_* and 
```

Comment:
may want to clarify if this is all unit commitment modes, or only integer.

Reply:
it works with linear relaxation.